Library/Homebrew/test/language/python_spec.rb

```diff
@@ -0,0 +1,124 @@
+  subject { described_class.new(formula, dir, "python") }
+
+  let(:dir) { Pathname.new(Dir.mktmpdir) }
+
+  let(:resource) { double("resource", stage: true) }
+  let(:formula_bin) { dir/"formula_bin" }
+  let(:formula) { double("formula", resource: resource, bin: formula_bin) }
+
+  after(:each) do
+    dir.rmtree
+  end
+
+  describe "#create" do
+    it "creates a virtual environment" do
+      expect(formula).to receive(:resource).with("homebrew-virtualenv").and_return(resource)
+      subject.create
+    end
+
+    specify "virtual environment creation is idempotent" do
+      expect(formula).to receive(:resource).with("homebrew-virtualenv").and_return(resource)
+      subject.create
+      FileUtils.mkdir_p dir/"bin"
+      FileUtils.touch dir/"bin/python"
+      subject.create
+      FileUtils.rm dir/"bin/python"
+    end
+  end
+
+  describe "#pip_install" do
+    it "accepts a string" do
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last).to eq("foo")
+
+        true
+      end
+
+      subject.pip_install "foo"
+    end
+
+    it "accepts a multi-line strings" do
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last(2)).to eq(["foo", "bar"])
+
+        true
+      end
+
+      subject.pip_install <<-EOS.undent
+        foo
+        bar
+      EOS
+    end
+
+    it "accepts an array" do
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last).to eq("foo")
+
+        true
+      end
+
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last).to eq("bar")
+
+        true
+      end
+
+      subject.pip_install ["foo", "bar"]
+    end
+
+    it "accepts a Resource" do
+      res = Resource.new("test")
+
+      expect(res).to receive(:stage).and_yield
+      expect(formula).to receive(:system) do |*args|
+        expect(args.first).to eq(dir/"bin/pip")
+        expect(args).to include("install")
+        expect(args.last).to eq(Pathname.pwd)
+
+        true
+      end
+
+      subject.pip_install res
+    end
+  end
+
+  describe "#pip_install_and_link" do
+    let(:bin) { dir/"bin" }
+    let(:dest) { formula.bin }
+
+    it "can link scripts" do
+      bin.mkpath
+
+      expect(bin/"kilroy").not_to exist
+      expect(dest/"kilroy").not_to exist
+
+      FileUtils.touch bin/"irrelevant"
+      bin_before = Dir.glob(bin/"*")
+      FileUtils.touch bin/"kilroy"
+      bin_after = Dir.glob(bin/"*")
+
+      expect(subject).to receive(:pip_install).with("foo")
+      expect(Dir).to receive(:[]).twice.and_return(bin_before, bin_after)
```

Comment:
I know it was there before, but I'm uneasy about this stub. Could we differentiate using arguments or something?